tests/Lykke.Job.BlockchainCashinDetector.IntegrationTests/Utils/CqrsActionAwaiter.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Lykke.Cqrs;
+using Lykke.Cqrs.Middleware;
+
+namespace Lykke.Job.BlockchainCashinDetector.IntegrationTests.Utils
+{
+    internal class CqrsActionAwaiter
+    {
+        private Dictionary<Type, EventWaitHandle> _processingDict =
+            new Dictionary<Type, EventWaitHandle>();
+
+        private SemaphoreSlim _lock = new SemaphoreSlim(1, 1);
```

Comment:
Why you need the lock here? Consider to use ConcurrentDictionary for _processingDict